app/models/step_by_step_page.rb

```diff
@@ -62,6 +62,22 @@ def unpublished_changes?
     has_draft? && has_been_published?
   end
 
+  def can_be_published?
+    has_draft?
+  end
+
+  def can_be_unpublished?
+    has_been_published?
+  end
+
+  def can_discard_changed?
```

Comment:
Very tiny nitpick: I think this should be called `can_discard_changes?`

Reply:
Good spot, thanks. It was meant to be `can_discard_changes?`. Fixed now